packages/requisitions/src/RequestRequisition/DetailView/AppBarButtons.tsx

```diff
@@ -18,11 +20,22 @@ export const AppBarButtonsComponent: FC<AppBarButtonProps> = ({
   onAddItem,
 }) => {
   const { OpenButton } = useDetailPanel();
-  const t = useTranslation('common');
+  const t = useTranslation('distribution');
+  const modalController = useToggle();
 
   return (
     <AppBarButtonsPortal>
       <Grid container gap={1}>
+        <ButtonWithIcon
+          Icon={<PlusCircleIcon />}
+          label={t('button.add-from-master-list')}
+          onClick={modalController.toggleOn}
+        />
+        <MasterListSearchModal
+          open={modalController.isOn}
+          onClose={modalController.toggleOff}
+          onChange={masterList => console.log(masterList)}
+        />
```

Comment:
as mentioned in the top level comment.. it's not a lot of code to add where you want it.. but it could be less 😁 

Though.. I suppose these should maybe lazily loaded 🤔

Reply:
yes, was wondering about generalising this - could just be one component which manages the state internally.
like you say though, this isn't much code - would be good to get this in place and then think about how it could be simplified for the consumer.